js/exporter/exceljs/exportDataGrid.js

```diff
@@ -1,62 +1,133 @@
-            currentColumnIndex++;
-        }
+    if(columns.length > 0) {
+        result.to.column += columns.length - 1;
 
-        if(columns.length > 0) {
-            result.to.column += columns.length - 1;
-        }
+        if(dataGrid.option("showColumnHeaders")) {
+            let headerRow = worksheet.getRow(result.to.row);
+            let currentColumnIndex = result.from.column;
 
-        if(options.excelFilterEnabled === true) {
-            worksheet.autoFilter = {
-                from: {
-                    row: result.from.row,
-                    column: result.from.column,
-                },
-                to: {
-                    row: result.to.row,
-                    column: result.to.column
-                }
-            };
-            worksheet.views = [{ state: 'frozen', ySplit: result.to.row }];
+            for(let i = 0; i < columns.length; i++) {
+                headerRow.getCell(currentColumnIndex).value = columns[i].caption;
+                currentColumnIndex++;
+            }
+
+            if(options.excelFilterEnabled === true) {
+                worksheet.autoFilter = {
+                    from: {
+                        row: result.from.row,
+                        column: result.from.column,
+                    },
+                    to: {
+                        row: result.to.row,
+                        column: result.from.column + columns.length - 1
+                    }
+                };
+                worksheet.views = [{ state: 'frozen', ySplit: result.to.row }];
+            }
+            result.to.row++;
         }
     }
 
-    result.to.row++;
-
     return new Promise((resolve) => {
-        dataGrid.getController("data").loadAll().then((items) => {
-            for(let i = 0; i < items.length; i++) {
+        var controllerRowsPromise = exportController._getAllItems();
+
+        controllerRowsPromise.done((controllerRows) => {
+            for(let rowIndex = 0; rowIndex < controllerRows.length; rowIndex++) {
                 let dataRow = worksheet.getRow(result.to.row);
+
+                dataRow.outlineLevel = _getRowOutlineLevel(exportController, controllerRows[rowIndex]);
+
                 let currentColumnIndex = result.from.column;
+                for(let cellIndex = 0; cellIndex < controllerRows[rowIndex].values.length; cellIndex++) {
+                    dataRow.getCell(currentColumnIndex).value = _getCellValue(exportController, controllerRows[rowIndex], cellIndex);
 
-                for(let j = 0; j < items[i].values.length; j++) {
-                    dataRow.getCell(currentColumnIndex).value = items[i].values[j];
                     currentColumnIndex++;
                 }
                 result.to.row++;
             }
-            result.to.row--;
+
+            if(result.to.row > topLeftCell.row) result.to.row--;
+
             resolve(result);
         });
     });
 }
 
+function _getCellValue(exportController, controllerRow, cellIndex) {
```

Comment:
Why you don't use DataProvider API instead implement same logic?

Reply:
The DataProvider is part of the old export. If we will changing the logic, it is better not to depend on the methods of the old export.